### ISSUE TITLE
Extend Travis checks

### ISSUE THREAD

**Issue by kisst** [awslabs/cfn-python-lint#663]
*cfn-lint version: (`cfn-lint --version`)*
0.14.0

*Description of issue.*

The Travis build check could be extended with `aws cloudformation validate-template` across the whole test/fixtures/templates/good folder, this was the good template are for sure good.

**Comment by cmmeyer** [comment]
Hey, thanks for the suggestion.

A primary design goal of cfn-lint is to keep the linter self-contained. It shouldn't have to rely on the API template validation to catch errors. The templates in the test directory are there as a mechanism to catch problems with the linter's rules -- we're not as concerned with whether they are actually valid and deployable as whole.  Validating these fixtures against the API would add significant time to the testing process without significant benefit.

**Comment by kisst** [comment]
Hey @cmmeyer 

We might have a misunderstanding, I do not suggested to run the calls against the AWS API at the time of code execution, but only for code testing, the the benefit is that errors like #662 can't happen, I believe the time to run the validate comparing to the whole tox run across all python version is not significant.

```
time find test/fixtures/templates/good -iname "*.yaml" -exec \
  aws cloudformation validate-template --template-body file://{} \;
real    0m42.568s
user    0m30.271s
sys     0m4.087s
```
While the whole tox test, is around 4-5 minutes, however by the look of it, the Travis builds are also disabled, (at least I can't see them https://travis-ci.org/awslabs/cfn-python-lint ) I also seen on PR on the code-coverage report runs now.

Considering all this, this would be quite a task overall, as many of the `good` templates are not valid cfn templates to start with, there is no ci/cd check to plug into an extra check, and even if all that given, still need some credentials for the aws api to make the validation calls.

However if neither above is considered, I'm happy to just close this issue and move on, but as I think that actually there would be a benefit so let's at least consider it.

**Comment by kddejong** [comment]
I may be missing something here but `aws cloudformation validate-template` wouldn't have caught the wrong value in `EngineVersion`.

```
Resources:
  AmazonMQConfiguration:
    Type: "AWS::AmazonMQ::Configuration"
    Properties:
      EngineType: "ACTIVEMQ"
      EngineVersion: "Bad 5.15.8"
      Name: "Configuration"
```
```
➜  aws cloudformation validate-template --template-body file://test3.yaml
{
    "Parameters": []
}
➜  echo $?                                                                                                                                 
0
```

**Comment by kddejong** [comment]
I would also add the `good` templates shouldn't be considered fully good.  They are designed to test the rules and the rules are different scenarios (code coverage plays here).  Usually it doesn't mean the entire template is `good`  That being said these probably could be restructured by rules and not under the idea of `bad` or `good`.

**Comment by kisst** [comment]
Oh, you are right, I somehow expected that to fail.... but I guess then this is why tools like the cfn-lint itself,  are that popular, I'm happy to close this ticket, and maybe open a different one about getting some ci/cd checks in place for PR's

**Comment by PatMyron** [comment]
Good templates that currently fail `aws cloudformation validate-template`:

```shell
cfn-python-lint/test/fixtures/templates/good $ ls -1 **/*.{json,yaml} | parallel 'echo {}; aws cloudformation validate-template --template-body file://{} | grep "ValidationError"'

decode/parsing.json
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: At least one Resources member must be defined.

core/config_cfn_lint.json
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: At least one Resources member must be defined.

functions_findinmap.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: [/Mappings/myMap/us-east-1] map keys must be strings; received numeric [32] instead

functions_cidr.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Parameter 'vpcSubnetPrefix' must match pattern ^(?:[0-9]{1,3}\\.){1}[0-9]{1,3}$

resource_properties.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: Unresolved resource dependencies [PartitionKeyName, TableName, PartitionKeyType, SortKeyName, SortKeyType] in the Resources block of the template

core/config_parameters.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: At least one Resources member must be defined.

core/config_cfn_lint.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: At least one Resources member must be defined.

functions/findinmap_keys.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: [/Mappings/CertificateMap] map keys must be strings; received numeric [2] instead

functions/relationship_conditions.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template error: every Fn::Equals object requires a list of 2 string parameters.

functions/sub_needed.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template error: instance of Fn::GetAtt references undefined resource GreetingLambda

mappings/name.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: At least one Resources member must be defined.

mappings/configuration.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: [/Mappings/RegionAccountToAZ/ap-northeast-1] map keys must be strings; received numeric [1.23456789E8] instead

mappings/key_name.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: At least one Resources member must be defined.

outputs/description.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: At least one Resources member must be defined.

parameters/default.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: At least one Resources member must be defined.

templates/description.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: At least one Resources member must be defined.
```

**Comment by kddejong** [comment]
The only trick here is that good may be misleading. It was an original technique that we should get rid of.  These templates are used to test specific rules so they are focused on certain scenarios.  This may mean they fail many other rules that aren’t being tested for that template. 

I have found scenarios that we didn’t cover in cfn-lint that validate did catch and we should add rules for that. Additionally we may wish to change how we organize these templates so that it isn’t that misleading
